Report/Template.Rmd

```diff
@@ -243,17 +251,267 @@ datasets = read.and.clean(dataset = "easySHARE_rel6_0_0.rda", wav = 99)
+
+```{r, eval = FALSE}
+# Function for calculation group shares per country
+group.share = function(df, y, agg.country = TRUE) {
+    if (agg.country) {
+        final = tapply(df[[y]], df$country, function(x) {
+            val = sum(x)/length(x)
+            return(val)
+        })
+    } else { # return total if country-level aggregation not specified
+        final = sum(df[[y]])/length(df[[y]])
+    }
+    return(final)
+}
+
+# Function for calculating labor participation rate per country / gender / age
+labor = function(df, .country, .gender, .age) {
+    dataset = df %>% 
+        filter(country == .country & gender == .gender & get(.age) == TRUE)
+    denom = nrow(dataset)
+    numer = with(dataset, sum(labor_ft | labor_pt))
+    perc  = round(numer/denom, 4)
+    
+    return(perc)
+}
+
+```
+
+args.list contains three sub-lists divided in countries (`c`), gender (`g`) and age (`a`). The list is passed as an argument to `labor`, which are simulatenously iterated over the sub-lists in parallel using `pmap`.
+Including `with` to specify the data frame on which we expand the grid upon, modifies the data frame in a new environment. This simplifies name assignment. 
+
+```{r, eval = FALSE}
+
+# use expand.grid to get all combinations of country / gender / age
+agegroups = names(df.out[, 4:6])
+args      = with(df.out, 
+                 expand.grid(levels(country), levels(gender), agegroups, 
+                             stringsAsFactors = FALSE))
+args.list = list(c = as.vector(t(args[1])), 
+                 g = as.vector(t(args[2])), 
+                 a = as.vector(t(args[3])))
+
+# Map over multiple arguments
+output   = pmap(args.list, function(c, g, a) labor(df.out, c, g, a))
+output.v = do.call("c", output)
+
+labor.part.share    = cbind(args, output.v)
+labor.part.share.df = spread(labor.part.share, Var3, output.v) %>% 
+    arrange(desc(Var2)) %>% 
+    set_colnames(c("Country", "Gender", 
+                   paste0("Age ", rep(c("50-54", "55-59", "60-64")))))
+```
+
+The results can be easily bound into a data frame and arranged using `dplyr`. The same package is used to subset by gender. Using the pipe operator `%>%` keeps our code compact and avoids unnecessary variable assignment and inefficient memory requirements.
+
+`vars` is actually a name vector that specifies whether labor force participation is full time, part time or not prevalent. Fetching it using `grep` from the cleaned data set makes it robust against name changes and ensures consistency. Conveniently, `grep` does not necessarily require regular expressions as an input. Instead, it is able to perform matches on simple strings and, in this case, returns the names that contain 'labor'. 
```

Comment:
This paragraph is not following a logical structre

Reply:
I'll specify the functions again. Then it will be clear.